src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -235,6 +259,8 @@ internal Conversion GetCollectionExpressionSpreadElementConversion(
             {
                 return Conversion.NoConversion;
             }
+            // This should be conversion from type rather than conversion
```

Comment:
Using conversion from expression rather than conversion from type is a bug. The one case that I'm aware of where this is observable is when the spread element type is `dynamic`. Unfortunately, fixing this would be a breaking change for that case.

For instance, the following is currently allowed but would be an error when using conversion from type:
```csharp
dynamic[] x = [1, 2, 3];
int[] y = [..x];
```

I could log a bug and reference that here if that seems useful.

Reply:
I've updated the comment to include the example.